operator/identity/crd_gc.go

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package identity
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/cilium/operator/metrics"
+	"github.com/cilium/cilium/operator/watchers"
+	"github.com/cilium/cilium/pkg/controller"
+	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/k8s/identitybackend"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+func (igc *GC) startCRDModeGC(ctx context.Context) error {
+	if igc.gcInterval != 0 {
+		igc.logger.WithField(logfields.Interval, igc.gcInterval).Info("Starting CRD identity garbage collector")
+
+		igc.mgr = controller.NewManager()
+		igc.mgr.UpdateController("crd-identity-gc",
+			controller.ControllerParams{
+				RunInterval:  igc.gcInterval,
+				DoFunc:       igc.gc,
+				NoErrorRetry: true,
+			})
+	}
+
+	return igc.wp.Submit("hearbeat-updater", igc.runHeartbeatUpdater)
```

Comment:
s/hearbeat/heartbeat/

Shouldn't this be only started when `gcInterval != 0`?

Reply:
I thought the same but avoided to do so because the previous implementation was starting the heartbeat-updater even when `gcInterval == 0`. Anyway, it seems to be something that matters only to the identities gc inner workings, so I guess it was unintended.
I'll add a separate commit to highlight the change from previous implementation.